File: frontend/src/types/interview.ts
/**
 * AI 面试相关类型定义
 */

/** 面试会话 */
export interface InterviewSession {
  id: number
  user_id: number
  title: string
  target_position: string
  interview_type: string
  difficulty: string
  question_count: number
  answer_mode: string
  status: string
  total_score: number | null
  report: Record<string, any> | null
  started_at: string | null
  ended_at: string | null
  created_at: string
}

/** 面试轮次 */
export interface InterviewTurn {
  id: number
  session_id: number
  question_index: number
  question: string
  answer_text: string | null
  answer_audio_url: string | null
  answer_duration_seconds: number | null
  score: number | null
  feedback: string | null
  suggestion: string | null
  created_at: string
  answered_at: string | null
}

/** 面试会话详情（含轮次） */
export interface InterviewSessionDetail extends InterviewSession {
  turns: InterviewTurn[]
}

/** 创建面试参数 */
export interface InterviewCreateParams {
  title?: string
  target_position: string
  interview_type?: string
  difficulty?: 'easy' | 'medium' | 'hard'
  question_count?: number
  answer_mode?: 'text' | 'audio' | 'mixed'
}

/** 提交回答参数 */
export interface InterviewAnswerParams {
  answer_text?: string
  answer_audio_url?: string
  answer_duration_seconds?: number
}

/** 面试报告响应 */
export interface InterviewReport {
  session_id: number
  total_score: number
  status: string
  summary: string
  turn_performance: TurnPerformance[]
  suggestions: string[]
  generated_at: string
}

/** 报告中单轮表现 */
export interface TurnPerformance {
  question_index: number
  question: string
  score: number
  feedback: string
  suggestion: string
}

/** 分页面试列表 */
export interface InterviewListResponse {
  items: InterviewSession[]
  total: number
  page: number
  page_size: number
}

File: backend/app/schemas/interview.py
"""
AI 面试相关 Pydantic Schema
"""

from datetime import datetime
from typing import Self

from pydantic import BaseModel, Field, model_validator


class InterviewSessionCreateRequest(BaseModel):
    """创建 AI 面试会话请求"""

    title: str | None = Field(default=None, min_length=1, max_length=256, description="面试标题")
    target_position: str = Field(..., min_length=1, max_length=128, description="目标岗位")
    interview_type: str = Field(default="technical", min_length=1, max_length=64, description="面试类型")
    difficulty: str = Field(default="medium", pattern="^(easy|medium|hard)$", description="难度")
    question_count: int = Field(default=5, ge=1, le=20, description="题目数量")
    answer_mode: str = Field(default="text", pattern="^(text|audio|mixed)$", description="回答方式")


class InterviewAnswerRequest(BaseModel):
    """提交面试回答请求"""

    answer_text: str | None = Field(default=None, max_length=20000, description="文字回答")
    answer_audio_url: str | None = Field(default=None, max_length=512, description="音频回答地址")
    answer_duration_seconds: int | None = Field(default=None, ge=0, description="回答时长（秒）")
    recording_id: int | None = Field(default=None, description="录音ID（用于获取转写文本）")

    @model_validator(mode="after")
    def validate_answer(self) -> Self:
        if not (self.answer_text and self.answer_text.strip()) and not self.answer_audio_url:
            raise ValueError("answer_text 和 answer_audio_url 至少提供一项")
        return self


class InterviewTurnResponse(BaseModel):
    """面试轮次响应"""

    id: int
    session_id: int
    question_index: int
    question: str
    answer_text: str | None = None
    answer_audio_url: str | None = None
    answer_duration_seconds: int | None = None
    score: float | None = None
    feedback: str | None = None
    suggestion: str | None = None
    created_at: datetime
    answered_at: datetime | None = None

    model_config = {"from_attributes": True}


class InterviewSessionResponse(BaseModel):
    """面试会话响应"""

    id: int
    user_id: int
    title: str
    target_position: str
    interview_type: str
    difficulty: str
    question_count: int
    answer_mode: str
    status: str
    total_score: float | None = None
    report: dict | None = None
    started_at: datetime | None = None
    ended_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class InterviewSessionDetailResponse(InterviewSessionResponse):
    """包含问答轮次的面试会话详情"""

    turns: list[InterviewTurnResponse] = Field(default_factory=list)


class InterviewTurnPerformance(BaseModel):
    """报告中的单轮表现"""

    question_index: int
    question: str
    score: float
    feedback: str
    suggestion: str


class InterviewReportResponse(BaseModel):
    """面试报告响应"""

    session_id: int
    total_score: float
    status: str
    summary: str
    turn_performance: list[InterviewTurnPerformance] = Field(default_factory=list)
    suggestions: list[str] = Field(default_factory=list)
    generated_at: datetime
